core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1238,8 +1239,21 @@ class KafkaApis(val requestChannel: RequestChannel,
     val metadataRequest = request.body[MetadataRequest]
     val requestVersion = request.header.apiVersion
 
+    val topicIds = metadataRequest.topicIds.asScala.toSet.filterNot(_ == Uuid.ZERO_UUID)
```

Comment:
Hey, with the new addition of gating topic IDs behind IBP 2.8, can we do a check of the interbroker protocol version in this method and if it is not at least 2.8, return with UNSUPPORTED_VERSION error with a message about topic IDs only in IBP 2.8 and above. There is a method in the 2.8 PR to easily get this information: `KafkaConfig.usesTopicId`

Here's the other PR for reference: https://github.com/apache/kafka/pull/9814
Here's the KIP with the updated error behavior: https://cwiki.apache.org/confluence/display/KAFKA/KIP-516%3A+Topic+Identifiers#KIP516:TopicIdentifiers-AdminandKafkaAdminClient

I will update the delete topics logic to follow this behavior as well.

Reply:
OK, I will add these check after 9814 merged.